invitationtracker/src/modules/discord/src/main/java/tracker/discord/InvitationTracker.java

```diff
@@ -62,10 +69,26 @@ private void handleMemberJoin(GuildMemberJoinEvent event, Guild e)
         {
             try
             {
-                e.retrieveInvites()
-                        .queue(fetchedInvites -> getUsedInvite(fetchedInvites, event.getGuild().getId())
-                                        .ifPresent(i -> storeInDB(i, event.getMember().getId()))
-                                , failure -> log.warn("Could not fetch invites", failure.getCause()));
+                e.retrieveInvites().queue(fetchedInvites -> {
+                    Optional<Invite> usedInvite = getUsedInvite(fetchedInvites, event.getGuild().getId());
+                    usedInvite.ifPresent(i -> {
+                        storeInDB(i, event.getMember().getId());
+                        GuildConfig guildConfig = SqlConnector.getGuildConfig(event.getGuild().getId());
+                        if (guildConfig != null) {
+                            if (guildConfig.getExecuteWhoisOnJoin() && guildConfig.getInternalWebhook() != null) {
+                                DateTimeFormatter fmt = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+                                sendWebhook(
+                                    guildConfig.getInternalWebhook(),
+                                    event.getMember().getAsMention() +
+                                     " (registered: `" + fmt.format(event.getMember().getTimeCreated()) +
+                                     "`) joined with invite <" + i.getUrl() +
+                                     "> (created `" + fmt.format(i.getTimeCreated()) +
+                                     "`) by " + i.getInviter().getAsMention() + "."
+                                );
+                            }
+                        }
```

Comment:
ist viel übersichtlicher und lesbarer wenn man den  Teil in so einem Block liebe in eine Methode auslagert.